TTCN3_Project/CANFilter/com/testingtech/playits/canfilter/ResponderTask.java

```diff
@@ -0,0 +1,76 @@
+package com.testingtech.playits.canfilter;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.net.Socket;
+import java.nio.charset.Charset;
+import java.util.TimerTask;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+public class ResponderTask extends TimerTask {
+
+  private String key;
+  private Car2XEntry car2xEntry;
+  private Socket socket;
+
+  /**
+   * Periodically sends a key-value pair encoded as a JSON String over a
+   * socket's output stream.
+   * 
+   * @param key
+   *            openXC key
+   * @param car2xEntry
+   *            corresponding value entry
+   * @param socket
+   *            TCP response channel
+   */
+  public ResponderTask(String key, Car2XEntry car2xEntry, Socket socket) {
+    this.key = key;
+    this.car2xEntry = car2xEntry;
+    this.socket = socket;
+  }
+
+  @Override
+  public void run() {
+    try {
+      JSONObject response = createResponse();
+      System.out.println("[ResponderTask] Sending " + response);
+      sendJSONObject(response);

```

Comment:
This is completely unnecessary. Qualifications are only necessary in case of nameclash. Cluttering everything with unnecessary this. qualifiation only makes the code unreadable.